Jenkinsfile

```diff
@@ -1,67 +1,145 @@
-        // Using the built container, run tests
-        customImage.inside("-e DEVITO_OPENMP=${DEVITO_OPENMP} -e OMP_NUM_THREADS=${OMP_NUM_THREADS}") {
-            sh "flake8 --builtins=ArgumentError ."
-            sh "py.test -n 2 -vs --cov"
-            // Additional seismic operator tests
-            if ( DEVITO_BACKEND!='yask' ) {
-                sh "python examples/seismic/benchmark.py test -P tti -so 4 -a -d 20 20 20 -n 5"
-                sh "python examples/seismic/benchmark.py test -P acoustic -a"
-                sh "python examples/seismic/acoustic/acoustic_example.py --full"
-                sh "python examples/seismic/acoustic/acoustic_example.py --constant --full"
-                sh "python examples/seismic/acoustic/gradient_example.py"
-                sh "python examples/misc/linalg.py mat-vec mat-mat-sum transpose-mat-vec"
-                sh "python examples/seismic/tti/tti_example.py -a"
-                sh "python examples/checkpointing/checkpointing_example.py"
-                // Test tutorial notebooks for the website using nbval
-                sh "py.test -vs --nbval examples/seismic/tutorials"
-                sh "py.test -vs --nbval examples/cfd"
-            }
-            // Now run coverage
-            sh "codecov"
-            // Now build documentation
-            sh "sphinx-apidoc -f -o docs/ examples/"
-            sh "sphinx-apidoc -f -o docs/ devito/ devito/yask/*"
-            sh "make -C docs/ html"
-        }
+def cleanWorkspace() {
+    sh "git clean -f -d"
+    sh "rm -rf ${WORKSPACE}/scratch"
+}
+
+def condaInstallDevito () {
+    sh 'conda env create -q -f environment.yml'
+    sh 'source activate devito ; pip install -e . ; pip install pytest-xdist ; conda list'
+    sh 'source activate devito ; flake8 --exclude .conda,.git --builtins=ArgumentError .'
+}
+
+def pipInstallDevito () {
+    sh "mkdir -p ${WORKSPACE}/lib/python3.6/site-packages/"
+    sh "python setup.py install --prefix=${WORKSPACE}"
+    sh 'bin/flake8 --exclude lib/python3.6/site-packages,.git --builtins=ArgumentError .'
+}
+
+def installYask () {
+    sh "mkdir -p $HOME/.ssh/"
+    sh """echo -e "Host github.com\n\tStrictHostKeyChecking no\n" >> $HOME/.ssh/config"""
+    sh "source activate devito ; conda install swig"
+    sh "mkdir ${WORKSPACE}/scratch"
+    dir ("${WORKSPACE}/scratch") { sh 'git clone https://github.com/opesci/yask.git' }
+    dir ("${WORKSPACE}/scratch/yask") { 
+        sh '''source activate devito
+              make compiler
+              make compiler-api
+              pip install -e .
+           '''
     }
 }
+
+def runPipTests() {
+    sh 'python setup.py test'
+}
+
+def runCondaTests() {
+    sh 'source activate devito ; py.test --cov devito tests/'
+}
+
+def runExamples () {
+    sh 'source activate devito ; python examples/seismic/benchmark.py test -P tti -so 4 -a -d 20 20 20 -n 5'
+    sh 'source activate devito ; python examples/seismic/benchmark.py test -P acoustic -a'
```

Comment:
have you attempted random failures here, to be 100% sure they are correctly caught ?

Reply:
No, good thought. I'll branch and do that.

Reply:
I've successfully got it to fail in flake8 and when in the example step.